backends/hip-ref/kernels/hip-ref-vector.hip.cpp

```diff
@@ -11,19 +11,19 @@
 //------------------------------------------------------------------------------
 // Kernel for set value on device
 //------------------------------------------------------------------------------
-__global__ static void setValueK(CeedScalar *__restrict__ vec, CeedInt size, CeedScalar val) {
-  int idx = threadIdx.x + blockDim.x * blockIdx.x;
+__global__ static void setValueK(CeedScalar *__restrict__ vec, CeedSize size, CeedScalar val) {
+  CeedSize idx = threadIdx.x + blockDim.x * blockIdx.x;
```

Comment:
Since [`dim3` consists of `uint32_t`](https://rocm.docs.amd.com/projects/HIP/en/latest/.doxygen/docBin/html/structdim3.html#dim3-struct-reference), we need to cast before the multiplication to avoid overflow prior to assignment to the larger type.
```suggestion
  CeedSize idx = threadIdx.x + (CeedSize)blockDim.x * blockIdx.x;
```
Same for the other occurrences.

Reply:
Good catch, thanks!  I fixed it in baa05712626079549b60ffa096dc74c6a7fb8160